ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -173,22 +173,38 @@ function groupNodes(nodes: any[], graph: dagre.graphlib.Graph) {
         groupedNodesArr.forEach((obj: {kind: string; nodeIds: string[]; parentIds: dagre.Node[]}) => {
             const {nodeIds, kind, parentIds} = obj;
             const groupedNodeIds: string[] = [];
+            const podGroupIds: string[] = [];
             nodeIds.forEach((nodeId: string) => {
                 const index = nodes.findIndex(node => nodeId === node.uid || nodeId === nodeKey(node));
-                if (index > -1) {
+                const graphNode = graph.node(nodeId);
+                if (!graphNode.podGroup && index > -1) {
                     groupedNodeIds.push(nodeId);
+                } else {
+                    podGroupIds.push(nodeId);
                 }
-                graph.removeNode(nodeId);
-            });
-            graph.setNode(`${parentIds[0].toString()}/child/${kind}`, {
-                kind,
-                groupedNodeIds,
-                height: NODE_HEIGHT,
-                width: NODE_WIDTH,
-                count: nodeIds.length,
-                type: NODE_TYPES.groupedNodes
             });
-            graph.setEdge(parentIds[0].toString(), `${parentIds[0].toString()}/child/${kind}`);
+            const reducedNodeIds = nodeIds.reduce((acc, aNodeId) => {
+                if (podGroupIds.findIndex(i => i === aNodeId) < 0) {
+                    acc.push(aNodeId);
+                }
+                return acc;
+            }, []);
+            if (groupedNodeIds.length === 1) {
```

Comment:
Can you explain this condition?

Reply:
This actually can be simplified.  It was to put back the node when it was removed on line 181 (in red) above.
This actually is a noop.

Reply:
Changes checked in

Reply:
Changed from what I had:

```
            if (groupedNodeIds.length === 1) {
                graph.node(groupedNodeIds.pop());
            } else {
                groupedNodeIds.forEach(n => graph.removeNode(n));
            }
```
to simply:

```
            if (groupedNodeIds.length > 1) {
                groupedNodeIds.forEach(n => graph.removeNode(n));
```

I'm removing the nodes after, so adding them back is not necessary.